placement/selector/non_mirrored.go

```diff
@@ -58,39 +58,15 @@ func (f *nonMirroredSelector) SelectAddingInstances(
 		return nil, err
 	}
 
-	candidateGroups := buildIsolationGroupMap(candidates)
-
-	existingGroups := buildIsolationGroupMap(p.Instances())
-
-	// If there is a isolation group not in the current placement, prefer the isolation group.
-	for r, instances := range candidateGroups {
-		if _, exist := existingGroups[r]; !exist {
-			// All the isolation groups have at least 1 instance.
-			return instances[:1], nil
-		}
+	if f.opts.AddAllCandidates() {
```

Comment:
could you also add some tests in placement/service/ to make sure this options is doing the right thing? For example when you add 10 instances but 5 of them already exist in the placement, verify that only the other 5 got added in the final result?